perf: disable hover when scrolling on web and desktop

hurali97 opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

This adds a prop `shouldHandleScroll` to **_Hoverable_** component, which decides whether we want to toggle `isHovered` state while scrolling or not. If `shouldHandleScroll` is set to `true` the **_Hoverable_** component will only toggle the `isHovered` state once the scroll is started and ended.

Since, react-native-web doesn't have [support](https://github.com/necolas/react-native-web/pull/1305) for `onScrollBeginDrag` and `onScrollEndDrag`, I moved forward with a temporary work around following [here](https://github.com/necolas/react-native-web/issues/1021#issuecomment-984151185).

The implemented solution looks like this:

- Added the temporary solution in _**BaseInvertedFlatList**_ component and we only use this when we are on web and desktop.
- We emit the scrolling state only once when scroll is started and ended, using the `DeviceEventEmitter`.
- In **_Hoverable_** component, we added an event listener only if `shouldHandleScroll` prop is `true` and listener is not already set.
- We then track the hover using `isHoveredRef` and use this once the scrolling is ended to update the `isHovered` state.

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/27214
PROPOSAL: https://github.com/Expensify/App/issues/27214 ( Proposal is in the issue description )


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

### Manual Test:

- Go to any Report which contains some messages.
- Start scrolling and notice that the hover and reaction bar doesn't display while you're scrolling.
- Once you stop the scroll, the hover and reaction bar will display.

- [X] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->
Same as tests

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- Go to any Report which contains some messages.
- Start scrolling and notice that the hover and reaction bar doesn't display while you're scrolling.
- Once you stop the scroll, the hover and reaction bar will display.

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>


https://github.com/Expensify/App/assets/47336142/4fdd5f2d-d97c-4221-9268-f0f10381efd6

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/47336142/40a314f4-ad41-421d-84d5-48ee1e58689b

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->

</details>

hurali97 committed:
perf: disable hover while scrolling on web and desktop

hurali97 committed:
Merge branch 'main' of https://github.com/hurali97/Expensify-App into perf/disable-hover-when-scrolling

hurali97 committed:
feat: handle scroll start

mountiny commented on package-lock.json:
```diff
@@ -27316,7 +27316,7 @@
     "node_modules/expensify-common": {
       "version": "1.0.0",
       "resolved": "git+ssh://git@github.com/Expensify/expensify-common.git#35bff866a8d345b460ea6256f0a0f0a8a7f81086",
-      "integrity": "sha512-O0KbaMljSFyoZXcxtx5B2qBL+n0md3wYOrArMVT8N0W+4wKncn+p8/vrmVlUMHe/vDloUbVmUZNdHwOZNWdx3w==",
```
I think this change is not needed here as there is no change to the package.json

Could you revert this?

abdulrahuman5196 replied:
+1

hurali97 replied:
Sure let me see !!

abdulrahuman5196 commented on src/CONST.ts:
```diff
@@ -2642,6 +2643,10 @@ const CONST = {
+    EVENTS: {
+        SCROLLING: 'scrolling',
+    },
+    IS_DESKTOP_AND_WEB: Platform.OS === 'web' || Platform.OS === 'macos' || Platform.OS === 'windows',
```
Is this really necessary, is it possible to use different files?
Note: Is this the abstraction we are talking in slack?

hurali97 replied:
Yes that's necessary as we want to call different `onScroll` method and also should `hoverable` component switch its behaviour ( for eg, show hover only once scroll stops ).

The reason we can't have different files is that we have the same implementation for web and desktop. So this flag tells us that whether we are on web or desktop.

Given that I should rename this key to `IS_DESKTOP_OR_WEB`.  👍🏻

> Note: Is this the abstraction we are talking in slack?

No. Abstraction is related to Event Emitters.

abdulrahuman5196 replied:
Given the issue urgency. Its fine with me to take it up in separate followup PR as well to have separate files and not have platform check.

cc: @mountiny If you are fine as well.

hurali97 replied:
Okay. So the two components which will go to platform specific files are `BaseInvertedFlatList` and `Hoverable`. The `Hoverable` component is already for web and desktop, since we have `index.native.ts` there. So we only need to refactor `BaseInvertedFlatList` and some parts of `Hoverable`.

mountiny replied:
We do prefer the different files per platform, @hurali97 are you saying here that this is not possible here?

abdulrahuman5196 commented:
@hurali97 There is lint failure as well.

hurali97 committed:
fix: prettier fixes

hurali97 committed:
revert: integrity in package-lock for expensify-common

hurali97 committed:
revert: expensify-common package-lock entry

hurali97 committed:
fix: upstream expensify-common package-lock entry

hurali97 commented:
@abdulrahuman5196 addressed all reviews apart from the platform specific one. I am up to tackle it in this PR, it shouldn't take much time. @mountiny what do you think ? 🤔

mountiny commented:
> Addressed all reviews apart from the platform specific one. I am up to tackle it in this PR, it shouldn't take much time. @mountiny what do you think ? 🤔

yes please, try to tackle it in this PR, we have deployed now so there is around 24 hours til the next deploy

hurali97 committed:
refactor: move to platform specific files

hurali97 committed:
revert: line break

hurali97 committed:
docs: add js docs for shouldHandleScroll prop

hurali97 commented:
@mountiny @abdulrahuman5196 Can you guys re-review at your ease, much appreciated. I have addressed the platform specific change ✅

mountiny commented:
thanks! great to get it in so quick, running the checks again

@abdulrahuman5196 can yu please check again now?

abdulrahuman5196 commented:
Sure will do in 1-2 hours

mountiny commented:
> How are we measuring the improvement on this? I could have missed some previous discussion as well. I am not sure if this change is making positive/negative impact apart from the logical theory?

I agree there is probably not an easy way to measure this but just from the experience on some longer chat with attachments and such, did it feel smoother?

hurali97 commented:
> Code Looks fine to me. I have 2 questions
> 
> 1. During testing I feel that the appearance of hover after the scroll is taking more time. We have kept 250 i assume, can we reduce it further so that the context menu icons comes faster.
> 2. How are we measuring the improvement on this? I could have missed some previous discussion as well. I am not sure if this change is making positive/negative impact apart from the logical theory?
> 
> @hurali97 @mountiny

Hey @abdulrahuman5196,

1. Yes we can reduce it further but it probably wouldn't affect that much as the main delay is due to the underlying component `RenderActionItem` as it's heavy to re-render. I have tried this by replacing `ReportActionItem` with a plain `View` and it was appearing relatively faster with same 250ms timeout. I think with https://github.com/Expensify/App/pull/26772 there will be improvements in the underlying renderItem.

2. Since this is improvement of a perceived performance, we measured this from a user perspective. You can follow the proposal for the before and after results.

mountiny commented:
@abdulrahuman5196 let me know if this is good to be merged, thanks!

hurali97 committed:
Merge branch 'Expensify:main' into perf/disable-hover-when-scrolling

mountiny commented:
Triggered a build for testing

abdulrahuman5196 commented:
@mountiny I am not exactly sure on the performance impact. Sometimes it felt like the changes caused negative impact. But I am not sure though, maybe could be due to dev or my machine. 

I will test again in the testing build. I think it's best to do some more testing before merging the change.

OSBotify commented:
Use the links below to test this build in android and iOS. Happy testing!

OSBotify commented:
:test_tube::test_tube: Use the links below to test this build in android and iOS. Happy testing! :test_tube::test_tube:
| android :robot:  | iOS :apple: |
| ------------- | ------------- |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/android/27236/index.html  | https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/27236/index.html  |
| ![Android](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/android/27236/index.html)  | ![iOS](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/27236/index.html)  |
| desktop :computer: | web :spider_web: |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/27236/NewExpensify.dmg  | https://27236.pr-testing.expensify.com  |
| ![Desktop](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/27236/NewExpensify.dmg)  | ![Web](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://27236.pr-testing.expensify.com)  |

mountiny commented:
> I will test again in the testing build. I think it's best to do some more testing before merging the change.

I agree with this!

abdulrahuman5196 commented:
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->




</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->




</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->





</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->





</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->




</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->




</details>

abdulrahuman5196 commented:
NAB: @hurali97 Kindly update `Offline Tests` as `Same as Tests` in author's checklist.

abdulrahuman5196 reviewed (approved):
Changes looks good and works well. Reviewers checklist is also complete.

Note: The scrolling is similar to that of staging IMO, but i also agree its hard to test performance objectively here. We can go ahead since the theory seems to be intact.

All yours. @mountiny 

🎀 👀 🎀
C+ Reviewed

hurali97 committed:
Merge branch 'main' of https://github.com/hurali97/Expensify-App into perf/disable-hover-when-scrolling

hurali97 committed:
fix: prettier fixes

hurali97 committed:
fix: capitalise

mountiny reviewed (approved):
Thanks everyone!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6245661756) to staging by https://github.com/mountiny in version: 1.3.72-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6274748223) to production by https://github.com/thienlnam in version: 1.3.72-11 🚀

platform | result
---|---
🤖 android 🤖|failure ❌
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅
